lightning/src/util/events.rs

```diff
@@ -1199,27 +1220,27 @@ impl MaybeReadable for Event {
 					let mut payment_hash = PaymentHash([0; 32]);
 					let mut payment_failed_permanently = false;
 					let mut network_update = None;
-					let mut all_paths_failed = Some(true);
 					let mut path: Option<Vec<RouteHop>> = Some(vec![]);
 					let mut short_channel_id = None;
 					let mut retry = None;
 					let mut payment_id = None;
+					let mut failure_opt = None;
 					read_tlv_fields!(reader, {
 						(0, payment_hash, required),
-						(1, network_update, ignorable),
+						(1, network_update, ignorable), // serialized in LDK versions prior to 0.0.114
 						(2, payment_failed_permanently, required),
-						(3, all_paths_failed, option),
 						(5, path, vec_type),
 						(7, short_channel_id, option),
 						(9, retry, option),
 						(11, payment_id, option),
+						(13, failure_opt, ignorable),
```

Comment:
The expanded macro code looks good to me. I think we'll only return `None` for the full event if the `MaybeReadable::read` returns `Ok(None)`, which only happens if the TLV type for the field is present.

Reply:
> maybe renaming the ignorable to ignorable_no_read or whatever?

That all sounds good, though I'd also be in favor of renaming `ignorable_*` to `upgradable_*`. Because I'd like to name the above variant `ignorable_required`, but that sounds like an oxymoron.